internal/pkg/utils/etcdhelper/assert.go

```diff
@@ -113,6 +138,14 @@ func AssertKVs(t assert.TestingT, client etcd.KV, expectedKVs []KV, ops ...Asser
 		}
 	}
 
+	// Dump actual state
+	if c.expectedStateFromFile != "" {
+		outDir := filepath.Join(filepath.Dir(c.expectedStateFromFile), ".out")              //nolint:forbidigo // no virtual FS
+		filePath := filepath.Join(outDir, filepath.Base(c.expectedStateFromFile)+".actual") //nolint:forbidigo // no virtual FS
+		assert.NoError(t, os.MkdirAll(outDir, 0o750))                                       //nolint:forbidigo // no virtual FS
+		assert.NoError(t, os.WriteFile(filePath, []byte(KVsToString(actualKVs)), 0o600))    //nolint:forbidigo // no virtual FS
+	}
+
```

Comment:
Actual state is saved to the `.out` directory, to be checked or copied to the expected file.